modules/core/src/main/java/org/apache/synapse/config/xml/TriggerErrorMediatorFactory.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.apache.synapse.config.xml;
+
+import org.apache.axiom.om.OMAttribute;
+import org.apache.axiom.om.OMElement;
+import org.apache.synapse.Mediator;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.mediators.v2.TriggerError;
+import org.jaxen.JaxenException;
+
+import javax.xml.namespace.QName;
+import java.util.Properties;
+
+/**
+ * Creates a trigger error mediator through the supplied XML configuration
+ * <p/>
+ * <pre>
+ * &lt;triggererror type="string" (errorMessage="string" | expression="expression")/&gt;
+ * </pre>
+ */
+public class TriggerErrorMediatorFactory extends AbstractMediatorFactory {
+
+    private static final QName ATT_ERROR_MSG = new QName("errorMessage");
+    private static final QName ATT_TYPE = new QName("type");
+    private static final QName TRIGGER_ERROR_Q
+            = new QName(XMLConfigConstants.SYNAPSE_NAMESPACE, "triggererror");
```

Comment:
`trigger-error` is more readable. WDYT?

Reply:
throw-error is much better IMO

Reply:
We have the following naming convention in mediators
<datamapper config="name" inputSchema="string" inputType="string" outputSchema="string" outputType="string"/> 
<jsontransform [schema="string"]>

Reply:
then throwError?

Reply:
+1